react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { activeIndex: 0 };
+    this.revolver = this.revolver.bind(this);
+    this.counter = this.counter.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
+    this.prevImg = this.prevImg.bind(this);
+  }
+
+  revolver() {
+    this.intervalId = setInterval(this.counter, 3000);
+  }
+
+  counter() {
+    this.setState({ activeIndex: this.state.activeIndex + 1 });
+    if (this.state.activeIndex === 4) {
+      this.setState({ activeIndex: 0 });
+    }
+  }
+
+  prevImg() {
+    this.setState({ activeIndex: this.state.activeIndex - 1 });
+    if (this.state.activeIndex <= 0) {
```

Comment:
You should do the `if` statement to determine the next index before setting state. State is async but sometimes it can happen fast so this can lead to bugs in your carousel.

Reply:
What's strange is that when I did this, everything on the screen would disappear after trying to switch from the first image to the last image of the carousel.  When I set it back to after `setState`, it worked again.

Reply:
The console would throw the error that it could not read imgUrl on line 47.